src/dataExplorer/components/DataExplorerPage.tsx

```diff
@@ -60,32 +60,23 @@ const DataExplorerPageHeader: FC = () => {
   }
 
   const handleRename = (name: string) => {
-    setResource({
-      ...resource,
-      data: {
-        ...resource.data,
-        name,
-      },
-    })
+    resource.data.name = name
+    save(resource?.language)
```

Comment:
This feels like an awkward api. `save` takes one argument and it's the language if the resource? Is there a better API, or a better name than `save`?

Reply:
> they are two separate things.

In this case, it's not a huge deal, but we shouldn't make a habit out of merging disparate changes. I realize our CI can be a pain. Working around that pain only makes the pain stay around.

> that's a byproduct of the patchwork design we have for all this work.

Cool. Let's make sure that issue is created and scheduled for next sprint before queuing this for merge.